core/controllers/dashboard_test.py

```diff
@@ -66,6 +75,349 @@ def test_logged_in_notifications_dashboard(self):
         self.logout()
 
 
+class DashboardStatisticsTest(test_utils.GenericTestBase):
+    # The model reads in each of these tests inside this class will not be
+    # replaced by calls to services, since these tests are written to carry out
+    # unit testing.
+
+    OWNER_EMAIL_1 = 'owner1@example.com'
+    OWNER_USERNAME_1 = 'owner1'
+    OWNER_EMAIL_2 = 'owner2@example.com'
+    OWNER_USERNAME_2 = 'owner2'
+
+    EXP_ID_1 = 'exp_id_1'
+    EXP_TITLE_1 = 'Exploration title 1'
+    EXP_ID_2 = 'exp_id_2'
+    EXP_TITLE_2 = 'Exploration title 2'
+
+    USER_SESSION = 'session1'
+
+    def setUp(self):
+        super(DashboardStatisticsTest, self).setUp()
+        self.signup(self.OWNER_EMAIL_1, self.OWNER_USERNAME_1)
+        self.signup(self.OWNER_EMAIL_2, self.OWNER_USERNAME_2)
+
+        self.owner_id_1 = self.get_user_id_from_email(self.OWNER_EMAIL_1)
+        self.owner_id_2 = self.get_user_id_from_email(self.OWNER_EMAIL_2)
+
+    def _record_start(self, exp_id, exp_version, state):
+        """Record start event to an exploration.
+        Completing the exploration is not necessary here since the total_plays
+        are currently being counted taking into account only the # of starts.
+        """
+        session_id = self.USER_SESSION
+        event_services.StartExplorationEventHandler.record(
+            exp_id, exp_version, state, session_id, {},
+            feconf.PLAY_TYPE_NORMAL)
+
+    def _rate_exploration(self, exp_id, num_ratings, rating_start):
+        """Create num_ratings ratings for exploration with exp_id,
+        of values starting from rating_start.
+        For example, num_ratings = 3 and rating_start = 2, then exploration
+        will be assigned the ratings 2,3,4.
+        """
+        # Each user id needs to be unique since each user can only give an
+        # exploration one rating.
+        user_ids = ['user%d' % i for i in range(num_ratings)]
+        rating = rating_start
+        self.process_and_flush_pending_tasks()
+        for user_id in user_ids:
+            rating_services.assign_rating_to_exploration(
+                user_id, exp_id, rating)
+            rating = rating + 1
+            if rating > 5:
```

Comment:
This wrapping seems a little odd, especially in a utility method which is supposed to be straightforward. I think it would be much clearer to just pass in a list of the ratings you want to add -- e.g. _rate_exploration(exp_id, [4]) to give a single rating of 4.